subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/VirtualPlatformState.java

```diff
@@ -57,30 +55,46 @@ void participatingModule(ModuleResolveState state) {
         }
     }
 
+    private String getForcedVersion() {
+        String version = null;
+        for (SelectorState selector : platformModule.getSelectors()) {
+            if (selector.isForce()) {
+                ComponentSelector requested = selector.getRequested();
+                if (requested instanceof ModuleComponentSelector) {
+                    String nv = ((ModuleComponentSelector) requested).getVersion();
+                    if (version == null || vC.compare(nv, version) < 0) {
+                        version = nv;
+                    }
+                }
+            }
+        }
+        return version;
+    }
+
     List<String> getCandidateVersions() {
+        String forcedVersion = getForcedVersion();
         ComponentState selectedPlatformComponent = platformModule.getSelected();
-        if (selectedPlatformComponent.getSelectionReason().isForced()) {
-            return Collections.singletonList(selectedPlatformComponent.getVersion());
-        }
-        List<String> sorted = Lists.newArrayListWithCapacity(participatingModules.size());
+        List<String> sorted = Lists.newArrayListWithCapacity(participatingModules.size() + 1);
+        sorted.add(selectedPlatformComponent.getVersion());
```

Comment:
❓ I wonder if the `forcedVersion` computed above does not need to be added as well when it is not `null` and different than the `selectedPlatformComponent.getVersion()`.

Reply:
I wonder too, but I didn't find any case where it does...